### PR TITLE
Improve web workflow error handling

### LINKED ISSUE #8109: Uploading a file that is too large via Web Workflow causes Hard Fault
### CircuitPython version

```python
8.2.0-beta.1-12-g0da62416a-dirty
```

### Code/REPL

```python
N/A
```

### Behavior

per @ladyada:

i don't think anyone tried uploading a file that is 'too large' - the board hardfaults

### Description

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by tannewt**
* Disable Nagle at the end of error messages so they are sent before the socket is closed.
* Correctly discard file contents when the PUT file is too large.
* Correctly reset file size after failure to setting it too large.
* Change diskinfo.json to be list of disks in preparation of supporting multiple disks.
* Bump version to 3 for the above change.
* Document diskinfo.json.

Fixes #8109

### CODE REVIEW

File: docs/workflows.md [hunk 1]
```diff
@@ -368,6 +368,31 @@ curl -v -L http://circuitpython.local/cp/devices.json
 }
 ```
 
+#### `/cp/diskinfo.json`
+
+Returns information about the attached disk(s). A list of objects, one per disk.
+
+* `root`: File sysstem path to the root of the disk.
```
Comment (dhalbert) [inline docs/workflows.md:375]:
```suggestion
* `root`: Filesystem path to the root of the disk.
```

File: supervisor/shared/web_workflow/web_workflow.c [hunk 2]
```diff
@@ -852,10 +891,16 @@ static void _reply_with_diskinfo_json(socketpool_socket_obj_t *socket, _request
...
-        "{\"free\": %d, "
+        "[{\"root\": \"/\", "
+        "\"free\": %d, "
         "\"block_size\": %d, "
-        "\"total\": %d}", free_clusters * block_size, block_size, total_size * block_size);
+        "\"writable\": %s, "
+        "\"total\": %d}]", free_clusters * block_size, block_size, writable, total_size * block_size);
 
     // Empty chunk signals the end of the response.
     _send_chunk(socket, "");
```
Comment (dhalbert) [inline supervisor/shared/web_workflow/web_workflow.c:906]: Does this force a flush too?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Typo and a question

File: supervisor/shared/web_workflow/web_workflow.c (hunk 2 above)
Comment (tannewt) [reply supervisor/shared/web_workflow/web_workflow.c:906]: Yes, `send_chunk("")` flushes. https://github.com/adafruit/circuitpython/pull/8667/files#diff-43f36ea6fd6b4d28f257e55f3f1f29a9890e72c88d7b56164d77ab5691b704ccR443

**Review by dhalbert — APPROVED** [review]
Thanks for this!

**Pull request merged**